pkg/providers/subnet/subnet.go

```diff
@@ -178,9 +180,9 @@ func (p *DefaultProvider) UpdateInflightIPs(createFleetInput *ec2.CreateFleetInp
 	defer p.Unlock()
 
 	// Process the CreateFleetInput to pull out all the requested subnetIDs
-	fleetInputSubnets := lo.Compact(lo.Uniq(lo.FlatMap(createFleetInput.LaunchTemplateConfigs, func(req *ec2.FleetLaunchTemplateConfigRequest, _ int) []string {
-		return lo.Map(req.Overrides, func(override *ec2.FleetLaunchTemplateOverridesRequest, _ int) string {
-			if override == nil {
+	fleetInputSubnets := lo.Compact(lo.Uniq(lo.FlatMap(createFleetInput.LaunchTemplateConfigs, func(req ec2types.FleetLaunchTemplateConfigRequest, _ int) []string {
+		return lo.Map(req.Overrides, func(override ec2types.FleetLaunchTemplateOverridesRequest, _ int) string {
+			if override == (ec2types.FleetLaunchTemplateOverridesRequest{}) {
```

Comment:
Is this state even possible? I think not based on how we build our overrides?

Reply:
This state should not be possible. We won't create launch templates unless we have all the necessary information now that we a have nodeclass readiness. I think we can drop this check